x-pack/legacy/plugins/apm/server/lib/errors/get_trace_errors_per_transaction.ts

```diff
@@ -33,7 +36,8 @@ export async function getTraceErrorsPerTransaction(
             { term: { [TRACE_ID]: traceId } },
             { term: { [PROCESSOR_EVENT]: 'error' } },
             { range: rangeFilter(start, end) }
-          ]
+          ],
+          must_not: [{ terms: { [ERROR_LOG_LEVEL]: excludedLogLevels } }]
```

Comment:
I'm not sure if it's better to list the log levels we want to treat as errors ("fatal", "error") or the ones we do not want to treat as errors like you did. Eg. "trace" is a another log level that we don't want to treat as an error so should be added to `excludedLogLevels`. Are there others we may have forgotten?

Reply:
Btw. I'm fine with this approach unless @beniwohli has any objections.

Reply:
I think a whitelist would be safer than a blacklist in this case, as the API doesn't limit the log level to a known list of levels, and many logging libraries allow custom log levels. In such a case, I'd rather not count the custom log level as error.

Reply:
> In such a case, I'd rather not count the custom log level as error.

Agree, let's take the whitelist approach then with `error`, `critical`, 'fatal' and `<none>` as indicators of an error.

Reply:
The reason I did it this way was I wasn't sure how to put `<none>` there. What would the query look like in that case?

Reply:
Do I literally use `<none>`?

Reply:
Something like this should do the trick

```js
const includedLogLevels = [ "fatal", "error", "critical" ];

// ...

{
  "query": {
    "bool": {
      "should": [
        { "bool": { "must_not": [{ "exists": { "field": ERROR_LOG_LEVEL } }] }},
        { "terms": { ERROR_LOG_LEVEL: includedLogLevels }
        }
      ]
    }
  }
}
```

`should` acts as 'OR' in this case.

Reply:
Updated in 217d25ba8a0.